ecs-cli/modules/cli/local/project/project.go

```diff
@@ -0,0 +1,214 @@
+		}
+	} else if filename != "" {
+		taskDefinition, err = p.readTaskDefinitionFromFile(filename)
+		if err != nil {
+			return err
+		}
+	} else if _, err := os.Stat(LocalInFileName); err == nil {
+	// Try reading local task-definition.json file by default
+		taskDefinition, err = p.readTaskDefinitionFromFile(LocalInFileName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if taskDefinition == nil {
+		return fmt.Errorf("Could not detect valid Task Definition")
+	}
+
+	p.taskDefinition = taskDefinition
+	return nil
+}
+
+func (p *localProject) readTaskDefinitionFromFile(filename string) (*ecs.TaskDefinition, error) {
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading task definition from %s: %s", filename, err.Error())
+	}
+
+	taskDefinition := ecs.TaskDefinition{}
+	err = json.Unmarshal(bytes, &taskDefinition)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing task definition JSON: %s", err.Error())
+	}
+
+	return &taskDefinition, nil
+}
+
+var newCommandConfig = func(context *cli.Context, rdwr config.ReadWriter) (*config.CommandConfig, error) {
+	return config.NewCommandConfig(context, rdwr)
+}
+
+// FIXME: NOTE this will actually read from either ARN or Task Definition family name
+func (p *localProject) readTaskDefinitionFromArn(arn string) (*ecs.TaskDefinition, error) {
+	rdwr, err := config.NewReadWriter()
+	if err != nil {
+		return nil, err
+	}
+
+	commandConfig, err := newCommandConfig(p.context, rdwr)
+	if err != nil {
+		return nil, err
+	}
+
+	ecsClient := ecsclient.NewECSClient(commandConfig)
+
+	return ecsClient.DescribeTaskDefinition(arn)
+}
+
+func (p *localProject) Convert() error {
+	// FIXME get secrets here, pass to converter?
+	data, err := converter.ConvertToDockerCompose(p.taskDefinition)
+
+	if err != nil {
+		return err
+	}
+
+	p.localBytes = data
+
+	return nil
+}
+
+func (p *localProject) Write() error {
+	// Will error if the file already exists, otherwise create
+
+	outputFileName := p.context.String(flags.LocalOutputFlag)
+	if outputFileName == "" {
+		outputFileName = LocalOutDefaultFileName
+	}
+	p.localOutFileName = outputFileName
+
+	out, err := os.OpenFile(outputFileName, os.O_WRONLY|os.O_CREATE|os.O_EXCL, LocalOutFileMode)
+	defer out.Close()
+
+	data := p.localBytes
+
+	if err != nil {
+		fmt.Printf("%s file already exists. Do you want to write over this file? [y/N]\n", outputFileName)
+
+		reader := bufio.NewReader(os.Stdin)
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			return fmt.Errorf("Error reading input: %s", err.Error())
+		}
+
+		formattedInput := strings.ToLower(strings.TrimSpace(input))
+
+		if formattedInput != "yes" && formattedInput != "y" {
+			return fmt.Errorf("Aborted writing compose file. To retry, rename or move %s", outputFileName) // TODO add force flag
+		}
+
+		// Overwrite local compose file
+		err = ioutil.WriteFile(outputFileName, data, LocalOutFileMode)
+		return err
+	}
+
+	_, err = out.Write(data)
```

Comment:
nit: Can we remove this statement by replacing it with the WriteFile above? It'll make the function slightly shorter

Reply:
there was something funky with how this was working with overwriting versus writing the output file for the first time that made me do it this way i think but let me try rejiggering this a little

Reply:
sure, no worries otherwise

Reply:
Okay that ended up being a lot of rejiggering but 3a73a58c should address this :)